tests/helper/helper_generic.go

```diff
@@ -177,6 +179,63 @@ func WatchNonRetCmdStdOut(cmdStr string, timeout time.Duration, check func(outpu
 	}
 }
 
+// RunCmdWithMatchOutputFromBuffer starts the command, and command stdout is attached to buffer.
+// we read data from buffer line by line, and if expected string is matched it returns true
+// It is different from WaitforCmdOut which gives stdout in one go using session.Out.Contents()
+// for commands like odo log -f which streams continuous data and does not terminate by their own
+// we need to read the stream data from buffer.
+func RunCmdWithMatchOutputFromBuffer(timeoutAfter time.Duration, matchString, program string, args ...string) (bool, error) {
```

Comment:
can we use Eventually instead like in this PR https://github.com/openshift/odo/pull/3619/files#diff-66f499c995fc950f403a939757588820R104?

Reply:
as mentioned in comments also, we cannot use `session.Out.Contents()`, it waits for command to run completely then give its all contents, `odo log -f` commands never terminates on its own, to have logs streaming it has to be running so have attached a buffer which streams logs from command output

Reply:
OK :+1: